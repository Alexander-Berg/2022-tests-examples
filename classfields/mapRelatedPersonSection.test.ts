import _ from 'lodash';

import type { CreditFormFields } from 'auto-core/react/components/common/CreditForm/types';
import personProfileMock from 'auto-core/react/dataDomain/credit/mocks/personProfile.mock';

import type { PersonProfile, PhoneEntity } from 'auto-core/types/TPersonProfile';

import mapRelatedPersonSection from './mapRelatedPersonSection';

it('дополнительный контакт - другого человека', () => {
    const initialValues = {} as CreditFormFields;
    const personProfile = _.pick(personProfileMock, [
        'related_persons',
        'phones',
    ]) as PersonProfile;

    mapRelatedPersonSection({ personProfile, initialValues });

    expect(initialValues).toMatchSnapshot();
});

it('дополнительный контакт - свой', () => {
    const initialValues = {} as CreditFormFields;
    const personProfile = _.pick(personProfileMock, [
        'phones',
    ]) as PersonProfile;

    const additionalPhoneEntity = {
        phone: '79168570422',
        phone_type: 'ADDITIONAL',
    } as PhoneEntity;

    personProfile.phones?.phone_entities.push(additionalPhoneEntity);

    mapRelatedPersonSection({ personProfile, initialValues });

    expect(initialValues).toMatchSnapshot();
});
