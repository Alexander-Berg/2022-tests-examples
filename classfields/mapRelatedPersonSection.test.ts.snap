// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`дополнительный контакт - другого человека 1`] = `
Object {
  "related_person/name": Object {
    "name": "Алексей",
    "patronymic": "Валерьевич",
    "surname": "Петров",
    "value": "Петров Алексей Валерьевич",
  },
  "related_person/phone": "+79168570429",
  "related_person/related_person_type": "RELATIVES",
}
`;

exports[`дополнительный контакт - свой 1`] = `
Object {
  "related_person/phone": "+79168570422",
  "related_person/related_person_type": "UNKNOWN_TYPE",
}
`;
